app/models/manageiq/providers/google/cloud_manager/refresh_parser.rb

```diff
@@ -314,6 +314,17 @@ def parse_instance(instance)
         flavor_uid       = parse_uid_from_url(instance.machine_type)
         flavor           = @data_index.fetch_path(:flavors, flavor_uid)
 
+        # If the flavor isn't found in our index, check if it is a custom flavor
+        # that we have to get directly
+        if flavor.nil? && flavor_uid.starts_with?("custom-")
```

Comment:
What about wrapping this in a function?  Something like

```ruby
# If the flavor is unknown (not all flavors are always listed), go query and add it
query_and_add_flavor(flavor_uid) if flavor.nil?
```

Then it'll work for any additional machine types that come up beyond custom types